Code/Source/Camera/CameraSensor.h

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include <Atom/Feature/Utils/FrameCaptureBus.h>
+#include <chrono>
+
+class Entity;
+
+namespace ROS2 {
+
+    //! Structure containing all information required to create the camera sensor
+    struct CameraSensorDescription {
+        //! Constructor to create the description
+        //! @param cameraName - name of the camera; used to differentiate the cameras in multi-camera setup
+        //! @param verticalFov - vertical field of view of camera sensor
+        //! @param width - camera image width in pixels
+        //! @param height - camera image height in pixels
+        CameraSensorDescription(const AZStd::string& cameraName, float verticalFov, int width, int height);
+        CameraSensorDescription() = delete;
+
+        const float verticalFieldOfViewDeg; //!< cameras vertical vield of view
+        const int width;                    //!< camera image width in pixels
+        const int height;                   //!< camera image height in pixels
+        const AZStd::string cameraName;     //!< camera nam eto differeniate the camera in mulit-camera setup
+
+        const float aspectRatio;              //!< camera image aspect ration; derived from widht and height
+        const AZ::Matrix4x4 viewToClipMatrix; //!< camera view to clip space transform matrix; derived from other parameters
+
+    private:
+        AZ::Matrix4x4 MakeViewToClipMatrix() const;
+        void validateParameters() const;
+    };
+
+    //! Class to create camera sensor using Atom renderer
+    //! It creates dedicated rendering pipeline for each camera
+    class CameraSensor {
+    public:
+        //! Function to initialize rendering pipeline for the camera sensor
+        //! @param cameraSensorDescription - camera sensor description used to create camera pipeline
+        void InitializePipeline(const CameraSensorDescription& cameraSensorDescription);
+
+        //! Function to deinitialize rendering pipeline for the camera sensor
+        void DestroyPipeline();
+
+        //! Function requiesting frame from rendering pipeline
+        //! @param cameraPose - current camera pose from which the rendering should take place
+        //! @param callback - callback function object that will be called when capture is ready
+        //!                   it's argument is readback structure containing, among other thins, captured image
+        void RequestFrame(const AZ::Transform& cameraPose, std::function<void(const AZ::RPI::AttachmentReadback::ReadbackResult& result)> callback);
+
+    private:
+        AZStd::vector<AZStd::string> m_passHierarchy;
+        AZ::RPI::RenderPipelinePtr m_pipeline;
+        AZ::RPI::ViewPtr m_view;
+
+        AZ::RPI::Scene* m_scene = nullptr;
+
+        bool m_isInitialized = false;
```

Comment:
I may not have the full picture, however this is strongly against my intuition of good practices. In C++, the sole existence of an object should establish its invariants (RAII). Using std::optional<CameraSensor> in the caller code may help to achieve this.